vcn_gateways.tf

```diff
@@ -174,14 +174,23 @@ resource "oci_core_route_table" "service_gw" {
 ###################
 # NAT Gateway (NGW)
 ###################
+resource "oci_core_public_ip" "nat_gateway_public_ip" {
+  count          = var.create_nat_gateway == true && var.nat_gateway_public_ip_id == "RESERVED" ? 1 : 0
+  compartment_id = var.compartment_id
+  lifetime       = "RESERVED"
+  freeform_tags  = var.freeform_tags
+  defined_tags   = var.defined_tags
+  display_name   = var.label_prefix == "none" ? var.nat_gateway_display_name : "${var.label_prefix}-${var.nat_gateway_display_name}"
+}
+
 resource "oci_core_nat_gateway" "nat_gateway" {
   compartment_id = var.compartment_id
   display_name   = var.label_prefix == "none" ? var.nat_gateway_display_name : "${var.label_prefix}-${var.nat_gateway_display_name}"
 
   freeform_tags = var.freeform_tags
   defined_tags  = var.defined_tags
 
-  public_ip_id = var.nat_gateway_public_ip_id != "none" ? var.nat_gateway_public_ip_id : null
+  public_ip_id = var.nat_gateway_public_ip_id != "none" ? var.nat_gateway_public_ip_id != "RESERVED" ?  var.nat_gateway_public_ip_id : join(",",oci_core_public_ip.nat_gateway_public_ip.*.id) : null
```

Comment:
can you explain this line for me please?

Also, switch the expected value for the variable to lower case.

Reply:
In existing code, `nat_gateway_public_ip_id` is a variable that takes  OCID of public IP created outside this module and assign that public IP to NAT gateway. I have introduced another value (RESERVED) for this variable

If nat_gateway_public_ip_id is set to ‘RESERVED”, the new code creates a public IP using `oci_core_public_ip `resource

`public_ip_id = var.nat_gateway_public_ip_id != "none" ? var.nat_gateway_public_ip_id != "RESERVED" ?var.nat_gateway_public_ip_id : join(",",oci_core_public_ip.nat_gateway_public_ip.*.id) : null`

In above line, the condition first checks if `var.nat_gateway_public_ip_id` is not set to “none” and if its a pass, then it checks whether that value is not “RESERVED”. If it passes both conditions, it picks up the OCID from var.nat_gateway_public_ip_id and assigns it to Nat Gateway. 

If var.nat_gateway_public_ip_id is not set to “none” and the value is “RESERVED”, it picks the OCID of the public IP created in code. The value comes from join(",",oci_core_public_ip.nat_gateway_public_ip.*.id)

If var.nat_gateway_public_ip_id is none, then it sets the value of public_ip_id to null and will allocate an ephemeral ip to Nat Gateway.

I'll change "RESERVED" to lowercase and update the pull request if you are ok with the proposed changes.